pymc3/variational/opvi.py

```diff
@@ -975,23 +979,25 @@ def __init_group__(self, group):
                 # self.ordering.size += None  # (np.prod(var.dshape[1:])).astype(int)
                 if self.local:
                     # XXX: This needs to be refactored
-                    shape = None  # (-1,) + var.dshape[1:]
+                    shape = (-1,) + var.dshape[1:]
```

Comment:
Perhaps you want to use the following:
```suggestion
                    shape = (-1,) + test_var.shape[1:]
```

Reply:
NB: What we're doing here is just a hack so that we can get concrete shape information.  We do that by using the initial sample values.  If the samples are ever expected to change shape, this approach won't work.